src/explore.sh

```diff
@@ -2,106 +2,145 @@
 # command it unifies the way that we search for things in the project, by
 # things, you can understand from files to messages in git log.
 
-include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_config_loader.sh"
 include "$KW_LIB_DIR/kwlib.sh"
 
+# Hash containing user options
+declare -gA options_values
+
 # The main entry point for the explore feature, if you want to add another
 # search mechanism, you probably wish to start by this function
-function explore()
+function explore_main()
 {
   local ret
 
-  explore_parser "$@"
-  ret="$?"
-
-  case "$ret" in
-    1)        # LOG
-      shift 1 # Remove 'log' string
-
-      if [[ "$#" -gt 2 && "$3" != "TEST_MODE" ]]; then
-        complain 'Too many parameters'
-        exit 22 # EINVAL
-      fi
-
-      explore_git_log "$@"
-      ;;
-    2)        # Use GNU GREP
-      shift 1 # Remove 'grep' string
-
-      if [[ "$#" -gt 2 && "$3" != "TEST_MODE" ]]; then
-        complain 'Too many parameters'
-        exit 22 # EINVAL
-      fi
-
-      explore_files_gnu_grep "$@"
-      ;;
-    3) # Search in directories controlled or not by git
-      shift 1
-
-      if [[ "$#" -gt 2 && "$3" != "TEST_MODE" ]]; then
-        complain 'Too many parameters'
-        exit 22 # EINVAL
-      fi
-
-      explore_all_files_git "$@"
-      ;;
-    4) # Search in files under git control
-      if [[ "$#" -gt 2 && "$3" != "TEST_MODE" ]]; then
-        complain 'Too many parameters'
-        exit 22 # EINVAL
-      fi
-
-      explore_files_under_git "$@"
-      ;;
-    *)
-      complain 'Invalid parameter'
-      exit 22 # EINVAL
-      ;;
-  esac
+  if [[ "$1" =~ -h|--help ]]; then
+    explore_help "$1"
+    exit 0
+  fi
+
+  parse_explore_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  flag="${options_values['TEST_MODE']}"
+  search="${options_values['SEARCH']}"
+  path="${options_values['PATH']}"
+
+  if [[ -n "${options_values['GIT_LOG']}" ]]; then
+    # LOG
+    explore_git_log "$search" "$path" "$flag"
+  fi
+
+  if [[ -n "${options_values['GNU_GREP']}" ]]; then
+    # Use GNU GREP
+    explore_files_gnu_grep "$search" "$path" "$flag"
+  fi
+
+  if [[ -n "${options_values['GIT_GREP']}" ]]; then
+    # Search in directories controlled or not by git
+    explore_all_files_git "$search" "$path" "$flag"
+  fi
+
+  if [[ -n "${options_values['FILE_UNDER_GIT']}" ]]; then
+    # Search in files under git control
+    explore_files_under_git "$search" "$path" "$flag"
+  fi
```

Comment:
I know you handle the multiple cases in the parse function, but I think adding a return at the end of each `if` will make the code more readable.